app/schema/api.py

```diff
@@ -243,6 +245,38 @@ class AddPriceOut(AddPriceIn):
     id: int
 
 
+class AddDiscountIn(BaseModel):
+    """AddDiscountIn."""
+
+    vlab_id: UUID
+    discount: Decimal
+    valid_from: AwareDatetime
+    valid_to: AwareDatetime | None = None
```

Comment:
For confirmation:
1. do we want to be able to apply a discount at any time during the day? (This is the same for regular prices, there is no enforcement on starting the price at midnight or at any hour during the day)
2.  do we want to allow `valid_from` to be in the past? (This is the same for regular prices, but maybe it's not a good idea, because it could create inconsistencies with the registered entries in the ledger. It could accept None to mean apply it immediately, for example)

These are mainly for discussion, we can open an issue for those.

Reply:
1. yes - any time works, the main use-case for this is when user upgrades an account to a "pro" or "premium" - we want to create a discount right away so that the user can immediately benefit from the subscription.

2. Good point, that doesn't make much sense (at least I can't see use-cases for that). I propose to add extra validation in a separate PR/iteration. Setting none is good idea, indeed.